src/quote/quote.service.ts

```diff
@@ -131,4 +137,93 @@ export class QuoteService implements OnModuleInit {
 
     await this.quoteRepository.save(quoteEntities);
   }
+
+  async getLatestPrice(deployment: Deployment, address: string, currencies: string[]): Promise<any> {
+    const enabledProviders = this.priceProviders[deployment.blockchainType].filter((p) => p.enabled);
+
+    let data = null;
+    let usedProvider = null;
+
+    for (const provider of enabledProviders) {
+      const shouldSkip = await this.shouldSkipProvider(deployment.blockchainType, address, provider.name);
+      if (shouldSkip) {
+        this.logger.log(`Skipping ${provider.name} due to previous failure for ${address}`);
+        continue;
+      }
+
+      try {
+        data = await this.fetchPriceFromProvider(provider.name, deployment, address, currencies);
+
+        if (
+          data &&
+          Object.keys(data).length > 0 &&
+          data[address.toLowerCase()] &&
+          Object.keys(data[address.toLowerCase()]).some((key) => key !== 'provider' && key !== 'last_updated_at')
```

Comment:
Minor thing: Maybe just my personal preference, but I'd suggest setting a variable with the 4th condition for improving clarity/readability: 
```
const hasDesiredFields = Object.keys(data[address.toLowerCase()]).some((key) => key !== 'provider' && key !== 'last_updated_at');
if (data &&
     Object.keys(data).length > 0 &&
     data[address.toLowerCase()] &&
    hasDesiredFields)
```
  
same condition is also on market-rate.controller.ts function